pkg/retry/azure_error.go

```diff
@@ -307,3 +326,48 @@ func HasStatusForbiddenOrIgnoredError(err error) bool {
 	}
 	return false
 }
+
+// ParseRawError parse the error message in the rawError and unmarshal it into RawErrorContainer
+func ParseRawError(rawError string) (*RawErrorContainer, error) {
+	reg := regexp.MustCompile(`^(?:[^{]*)([\s\S]*)$`)
+	matches := reg.FindStringSubmatch(rawError)
+	if len(matches) != 2 {
+		klog.V(4).Infof("skipping parsing because the format of the raw error message %q is not the expected one")
+		return nil, nil
+	}
+
+	rawErrorMap := make(map[string]*RawErrorContainer)
+	err := json.Unmarshal([]byte(matches[1]), &rawErrorMap)
+	if err != nil {
+		return nil, err
+	}
+
+	return rawErrorMap["error"], nil
+}
+
+// IsErrorLoadBalancerInUseByVirtualMachineScaleSet determines if the Error is
+// LoadBalancerInUseByVirtualMachineScaleSet
+func IsErrorLoadBalancerInUseByVirtualMachineScaleSet(rawError string) bool {
+	return strings.Contains(rawError, "LoadBalancerInUseByVirtualMachineScaleSet")
+}
+
+// GetVMSSMetadataByRawError gets the vmss name by parsing the error message
+func GetVMSSMetadataByRawError(rawError string) (string, string, error) {
+	if !IsErrorLoadBalancerInUseByVirtualMachineScaleSet(rawError) {
+		return "", "", nil
+	}
+
+	rawErrorInfo, err := ParseRawError(rawError)
+	if err != nil {
+		klog.Warningf("GetVMSSMetadataByRawError: failed to parse raw error: %v", err)
+		return "", "", nil
+	}
+
+	reg := regexp.MustCompile(`.*/subscriptions/(?:.*)/resourceGroups/(.*)/providers/Microsoft.Compute/virtualMachineScaleSets/(.+).`)
+	matches := reg.FindStringSubmatch(rawErrorInfo.Message)
+	if len(matches) != 3 {
+		return "", "", fmt.Errorf("GetVMSSMetadataByRawError: the error message %q is not valid", rawErrorInfo.Message)
```

Comment:
s/GetVMSSMetadataByRawError: the error message %q is not valid/couldn't find a VMSS resource Id from error message %s/